src/snarky/parties-helpers.ts

```diff
@@ -52,183 +56,115 @@ function asFieldsAndAux<T, TJson>(typeData: Layout, customTypes: CustomTypes) {
   };
 }
 
-function toJson(typeData: Layout, value: any, customTypes: CustomTypes): any {
```

Comment:
lots of refactoring here -- see the new `mapReduce` function at the bottom of the file, which abstracts what most of the functions here were doing!